src/h2olog/h2olog.cc

```diff
@@ -190,21 +217,26 @@ int main(int argc, char **argv)
     }
 
     int debug = 0;
+    int list_and_exit = 0;
     FILE *outfp = stdout;
     std::vector<std::string> event_type_filters;
     std::vector<std::string> response_header_filters;
     int c;
     pid_t h2o_pid = -1;
-    while ((c = getopt(argc, argv, "hdp:t:s:w:")) != -1) {
+    while ((c = getopt(argc, argv, "hdp:t:s:w:l")) != -1) {
         switch (c) {
         case 'p':
             h2o_pid = atoi(optarg);
             break;
         case 't':
-            event_type_filters.push_back(optarg);
+            for (const auto &item : split_str(',', optarg)) {
+                event_type_filters.push_back(item);
```

Comment:
The code should check if the given name is valid.

I think that the code would become straightforward if we do the following:
* Obtain the list of supported probes before calling `getopt`, and also create an empty list of probes to be registered.
* When receiving `-t` option, check if the specified name exists in the list of supported probes. If it does, add it to the list of probes to be registered. Otherwise, exit with an error.
* After iterating through all the options, copy the list of supported probes to the list of the probes to be registered, if the latter list is empty.

Reply:
Sounds nicer for user experience. Fixed at dd967cb.